float/src/main/java/com/novoda/floatschedule/convert/SheetsFloatGithubUserConverter.java

```diff
@@ -0,0 +1,74 @@
+package com.novoda.floatschedule.convert;
+
+import com.novoda.github.reports.sheets.network.SheetsServiceClient;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import rx.Scheduler;
+import rx.schedulers.Schedulers;
+
+import static com.novoda.floatschedule.convert.FloatNameFilter.byFloatName;
+import static com.novoda.floatschedule.convert.GithubUsernameFilter.byGithubUsername;
+import static com.novoda.floatschedule.convert.NoMatchFoundException.noMatchFoundExceptionFor;
+
+public class SheetsFloatGithubUserConverter implements GithubUserConverter {
+
+    private final Map<String, String> floatToGithubUser;
+    private final Scheduler scheduler;
+    private final SheetsServiceClient sheetsServiceClient;
+
+    public static SheetsFloatGithubUserConverter newInstance() {
+        Scheduler scheduler = Schedulers.immediate();
+        SheetsServiceClient sheetsServiceClient = SheetsServiceClient.newInstance();
+        return new SheetsFloatGithubUserConverter(sheetsServiceClient, scheduler);
+    }
+
+    SheetsFloatGithubUserConverter(SheetsServiceClient sheetsServiceClient, Scheduler scheduler) {
+        floatToGithubUser = new HashMap<>();
+        this.scheduler = scheduler;
+        this.sheetsServiceClient = sheetsServiceClient;
+    }
+
+    @Override
+    public List<String> getGithubUsers() throws Exception {
+        readIfNeeded();
+        return floatToGithubUser.values()
+                .stream()
+                .collect(Collectors.toList());
+    }
+
+    private void readIfNeeded() {
+        if (!floatToGithubUser.isEmpty()) {
+            return;
+        }
+        sheetsServiceClient.getEntries()
+                .subscribeOn(scheduler)
+                .subscribe(entry -> floatToGithubUser.put(entry.getTitle(), entry.getContent()));
+    }
+
+    @Override
+    public String getFloatUser(String githubUsername) throws Exception {
+        readIfNeeded();
```

Comment:
Shouldn't this bit be part of the stream chain?
As of now it's fine because the implementation use the immediate Scheduler but if we use something else we're gonna have a race condition.